src/shogun/mathematics/linalg/ratapprox/logdet/opfunc/DenseMatrixExactLog.cpp

```diff
@@ -11,7 +11,7 @@
 
 #include <shogun/mathematics/eigen3.h>
 
-#if EIGEN_VERSION_AT_LEAST(3,1,0)
+#if EIGEN_VERSION_AT_LEAST(3,1,0) && (EIGEN_WORLD_VERSION == 3 && EIGEN_MAJOR_VERSION == 2 && EIGEN_MINOR_VERSION < 6 )
```

Comment:
Can you use the CMake function `VERSION_LESS`
See e.g. `examples/meta/cpp/CMakeLists.txt`
It is a bit cleaner